x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -86,14 +82,10 @@ public LogicalPlan optimize(LogicalPlan verified) {
         Batch operators = new Batch("Operator Optimization",
                 new ConstantFolding(),
                 // boolean
-                new BooleanSimplification(),
+                new EqlBooleanSimplification(),
                 new LiteralsOnTheRight(),
                 new BinaryComparisonSimplification(),
                 new BooleanFunctionEqualsElimination(),
-                // needs to occur before BinaryComparison combinations
-                new PropagateEquals(),
-                new PropagateNullable(),
-                new CombineBinaryComparisons(),
```

Comment:
Interesting to see how multi-value can impact the single-value operations too. Might be worth a before/after performance analysis at some point, if it's worth adding an EQL mode to split multi- vs single-value querying.

Reply:
I am not convinced this is a yes/no decision on multi vs single-value querying. At the moment, EQL is using in some places multi-value querying (through ES/Lucene native query support) and single value querying (where Painless is used) in other places, even though it should be single value everywhere (we do document we do not support multi-value querying). Your suggestion of having a switch between multi vs single value querying means that we need to go against Elasticsearch's own way of doing things by creating Painless scripts everywhere.